src/coreclr/jit/lsra.cpp

```diff
@@ -2309,12 +2308,43 @@ BasicBlock* LinearScan::findPredBlockForLiveIn(BasicBlock* block,
 
     if (block->bbPreds == nullptr)
     {
+        assert((block != compiler->fgFirstBB) || (prevBlock != nullptr));
+        JITDUMP("\n\nNo predecessor; ");
+
+        // The shared throw blocks do not have predecessor because no block's bbJumpDest points to
+        // them. For such blocks, we want to return the fact that predecessor is indeed null instead
+        // of returning prevBlock. Returning prevBlock will be wrong, because LSRA would think that
+        // the variable is live in registers based on the lexical flow, but that won't be true from
+        // control flow perspective.
+        // Example:
+        //
+        // IG05:
+        //      ...         ; V01 is in 'rdi'
+        //      JNE IG07
+        //      ...
+        // IG06:
+        //      ...
+        //      ...         ; V01 is in 'rbx'
+        //      JMP IG08
+        // IG07:
+        //      ...         ; LSRA thinks V01 is in 'rbx' if IG06 is set as previous block of IG07.
+        //      ....
+        //      CALL CORINFO_HELP_RNGCHKFAIL
+        //      ...
+        // IG08:
+        //      ...
+        //      ...
+        if (block->bbJumpKind == BBJ_THROW && ((block->bbFlags & BBF_SHARED_THROW) != 0))
```

Comment:
I was thinking if I should add a new flag `BBF_SHARED_THROW` but I was not certain if `BBF_DONT_REMOVE` for a `throw` block would capture the case that I wanted to check.

Reply:
BBF_DONT_REMOVE is used on many blocks; checking that as well as BBJ_THROW would work, but might possibly pick up additional cases. You could check diffs to see, perhaps. Adding the flag seems reasonable.

Reply:
The shared throw helpers are special in that they don't have any live-in vars of their own (that is part of how we get away with not giving them actual preds). So you could perhaps instead check that the block's use set is empty and that it is also BBJ_THROW and avoid special up front analysis and block tagging.

It seems odd to me that LSRA doesn't get rid of unreachable blocks and instead tolerates them.

Reply:
Just checking if the block is `BBJ_THROW` without any predecessor works, so I don't need `BBF_SHARED_THROW` anymore.